Rework the client protocol module.

knightburton opened the pull request:
- Apply the new eslint rules to the module.
- Rework the class definition based on es6.

IoT.jsCode-DCO-1.0-Signed-off-by: Imre Kiss kissi.szeged@partner.samsung.com

daeyeon commented on src/app/client-protocol.js:
```diff
@@ -18,928 +18,935 @@ import Connection from './client-connection';
+  /**
+   * Packages sent by the server to the client.
+   */
+  get SERVER_PACKAGE() {
```
Since DebuggerClient isn't a singleton class this may waste memory whenever instantiating the class. We could instead define the static object outside of the class and refer it in the getter.

e.g) 
```
let somthing = {
  JERRY_DEBUGGER: 1,
};

class Test {
  constructor() {
  }
  get Package() {
    return somthing;
  }
}
```

daeyeon commented on src/app/client-protocol.js:
```diff
@@ -18,928 +18,935 @@ import Connection from './client-connection';
-/**
- * Getters and setters.
- */
+export default class DebuggerClient {
```
Unlike other files, this filename, `client-protocol.js`, looks a bit difficult to infer its classname, `DebuggerClient`.

knightburton replied:
Well, you are right, what do you prefer? Shoul we change the filename or the classname?

daeyeon replied:
Anything you want is fine for me if the pair, classname and its filename, is similar enough to easily find.

knightburton replied:
Well, I think the `Protocol` name for the class would be great, but I think this should be done in an other commit after the rework done (or at least the relevant modules) because I think that we should use a proper name to the instance object too. This means that I would like to replace the `debuggerObj` instance name to `protocol` in tha main.js, connection module etc... What do you think about this?

daeyeon replied:
Thanks for asking my suggestion. Once seeing `SERVER_PACKAGE` , `CLIENT_PACKAGE` and its occurrences, then `Protocol` for the class seems to make sense. However, the class properties like `_newFunctions` and `_mode` seem a model data for `DebuggerClient` rather than anything related to the protocol definitions. So `Protocol` for the class name is a bit confusing. IMHO, we could get the protocol stuffs out of `DebuggerClient` as the following.

```javascript
// client-debugger.js
export const PROTOCOL =  {
    SERVER: {
        ...
    },
    CLIENT: {
        ...
    },
}

export default function DebuggerClient(...) {
   ...
}

// client-XXX.js
import { PROTOCOL } from 'client-debugger';
...
if (message[0] === PROTOCOL.CLIENT.JERRY_DEBUGGER_CONTINUE) {
}
...
```
I agree to refator the naming in another PR.

daeyeon commented on src/app/client-protocol.js:
```diff
@@ -18,928 +18,935 @@ import Connection from './client-connection';
+        let sourceName = func.sourceName;
+
+        if (sourceName === line[1]
+            || sourceName.endsWith('/' + line[1])
```
Regarding linebreak style for operators, I found different style in https://github.com/knightburton/iotjscode/blob/494d89e85d6a681231b3afb47869cd603215749c/src/app/main.js#L965

Do you mind to add https://eslint.org/docs/rules/operator-linebreak with default, `after`?

knightburton replied:
I do not mind, and as can I see the PR is landed :)

daeyeon replied:
Thanks for your understanding 👍

glistening commented on src/app/client-protocol.js:
```diff
@@ -18,928 +18,935 @@ import Connection from './client-connection';
+  /**
+   * Packages sent by the server to the client.
+   */
+  get SERVER_PACKAGE() {
```
@knightburton Do you know what rule translated the `const` to `get`? Personally I think `const` is straightforward, I would like to know the reason of this translation.

knightburton replied:
Well, these lists are logical part of the debugger class and they used a lot, therefore I just put these inside the class and I did not wanted to add these list to the class with the `this.something = something;` terminology. So basicaly this is just a personal idea but if you want to use these and other way I am openminded. Should I write this back to const at outside?

knightburton commented:
@daeyeon I've updated the PR, but I do not rebased the commit yet, because I would like to hear your opinion about it. I've changed a lots of code at this time based on your suggestion, I did not separate the naming because this way (I've only renamed the file at the end and exported the constant lists) it is not a big change if we look at it all.

daeyeon commented:
@knightburton I'm fine with your commit in progress. Please combine them into a single commit once finished.

knightburton commented:
@daeyeon I've finished the update.

yichoi commented:
LGTM
